nimbus-core/src/main/java/com/antheminc/oss/nimbus/domain/cmd/exec/internal/search/DefaultSearchFunctionHandlerLookup.java

```diff
@@ -121,11 +130,41 @@ private R getDynamicParamValues(LookupSearchCriteria lookupSearchCriteria, Class
 					paramValues.add(new ParamValue(code.toString(), label.toString()));
 				}
 			}
-			return (R)paramValues;
+			return sortIfApplicable(paramValues, mConfig, cmd);
 		}
 		catch(Exception ex) {
-			throw new FrameworkRuntimeException("Failed to parse property - code: "+list.get(0)+" and label: "+list.get(1), ex);
+			throw new FrameworkRuntimeException("Failed to parse property - code: "+list.get(0)+" and label: "+list.get(1)+" for command: "+cmd, ex);
 		}
 	}
 	
+	/**
+	 * In memory sorting of the param values
+	 */
+	private R sortIfApplicable(List<ParamValue> paramValues, ModelConfig<?> mConfig, Command cmd) {
+		String orderBy = cmd.getFirstParameterValue(Constants.SEARCH_REQ_ORDERBY_MARKER.code);
+		
+		if(StringUtils.isBlank(orderBy) || StringUtils.startsWith(orderBy, findRepoAlias(mConfig)+"."))
+			return (R) paramValues;
+		
+		int index = StringUtils.lastIndexOf(orderBy, propertyDelimiter);
+		
+		if(index == -1)
+			throw new IllegalStateException("Invalid orderby clause for StaticCodeValue search for a command "+cmd+" . It must follow the pattern \"property.direction\" e.g. code.asc() or name.firstName.desc()");
+		
+		String property = StringUtils.substringBeforeLast(orderBy, propertyDelimiter);
+		String direction = StringUtils.substringAfterLast(orderBy, propertyDelimiter);
+		
+		try {
+			if(StringUtils.equalsAnyIgnoreCase(direction, Constants.SEARCH_REQ_ORDERBY_DESC_MARKER.code))
+				CollectionsTemplate.sortSelfReverse(paramValues, Comparator.comparing(pv -> expressionEvaluator.getValue(property, pv, String.class)));
+			else
```

Comment:
If someone were to give `...orderby=staticCodeValue.paramCode.reverse()`, the sort would still occur but in ascending order. I think we should also explicitly look for `.asc()` (or whatever the desired keyword is) to avoid any confusion that might occur.

Reply:
We should absolutely put `asc()` and `desc()` in the documentation. 😄 

I agree that these should be the only two options to match what we have in MongoDB sorting. But if the user does send `.reverse()` the framework would not throw an exception as this is written, right? It would just send the list back ascending order. If an unknown sort direction is given, I'd think we would fail fast and throw an exception.

Reply:
@lopezton-anthem added the exception check. please review.

Reply:
👍